tests/phpunit/tests/rewrite/oldDateRedirect.php

```diff
@@ -66,6 +67,36 @@ public function test_old_date_redirect() {
 		$this->assertSame( $permalink, $this->old_date_redirect_url );
 	}
 
+	public function test_old_date_redirect_uses_cache_for_second_query() {
+		global $wpdb;
+		$old_permalink = user_trailingslashit( get_permalink( self::$post_id ) );
+
+		$time = '2004-01-03 00:00:00';
+		wp_update_post(
+			array(
+				'ID'            => self::$post_id,
+				'post_date'     => $time,
+				'post_date_gmt' => get_gmt_from_date( $time ),
+			)
+		);
+
+		$permalink = user_trailingslashit( get_permalink( self::$post_id ) );
+
+		$this->go_to( $old_permalink );
+		wp_old_slug_redirect();
+		$this->assertSame( $permalink, $this->old_date_redirect_url );
+
+		$query_count = $wpdb->num_queries;
+
+		$this->go_to( $old_permalink );
+		wp_old_slug_redirect();
+		$this->assertSame( $permalink, $this->old_date_redirect_url );
+
+		$this->assertSame( $query_count, $wpdb->num_queries );
```

Comment:
Check if these are failing due to the `go_to()` -- if it uses `WP_Query` then an additional query is expected.